src/main/java/org/sagebionetworks/bridge/validators/StudyValidator.java

```diff
@@ -135,16 +141,21 @@ public void validate(Object object, Errors errors) {
                 errors.rejectValue("eventId", CANNOT_BE_BLANK);
             } else if (!customEvent.getEventId().matches(BRIDGE_RELAXED_ID_PATTERN)) {
                 errors.rejectValue("eventId", BRIDGE_RELAXED_ID_ERROR);
-            } else {
-                uniqueIds.add(customEvent.getEventId());    
             }
+            uniqueIds.add(customEvent.getEventId());    
             if (customEvent.getUpdateType() == null) {
                 errors.rejectValue("updateType", CANNOT_BE_NULL);
             }
             errors.popNestedPath();
         }
         if (uniqueIds.size() > 0 && (uniqueIds.size() != study.getCustomEvents().size())) {
-            errors.rejectValue(CUSTOM_EVENTS_FIELD, "cannot contain duplidate event IDs");
+            errors.rejectValue(CUSTOM_EVENTS_FIELD, "cannot contain duplicate event IDs");
+        }
+        if (protectedCustomEventIds != null && !uniqueIds.containsAll(protectedCustomEventIds)) {
+            protectedCustomEventIds.removeAll(uniqueIds);
+            errors.rejectValue(CUSTOM_EVENTS_FIELD, 
+                    String.format("cannot remove custom events currently used in a schedule: [%s]",
```

Comment:
Minor: you can remove the brackets in the string

Reply:
Sounds good, done.